metrics-delivery-endpoint-impl/src/main/java/eu/openaire/mas/delivery/provider/PrometheusMetricsProvider.java

```diff
@@ -76,6 +66,33 @@ public MetricEntry deliver(String resourceId, String metricId, String from, Stri
         
     }
 
+    private MetricEntry runQuery(String resourceId, String metricId, String query) {
+	try {
+	    if (query.startsWith(SPEL_PREFIX)) {
+		float value = runSpEL(query.substring(SPEL_PREFIX.length()));
+		return new MetricEntry(resourceId, metricId, value);
+	    }
+	    // FIXME it is just a sample, include "from" and "to" params in querying
+	    VectorResponse resp = prometheusClient.query(query);
+	    if (STATUS_SUCCESS.equals(resp.getStatus())) {
+		float value = resp.getData().getResult().get(0).getValue().get(1);
+		return new MetricEntry(resourceId, metricId, value);
+	    } else {
+		throw new RuntimeException(String.format("invalid status: %, full response: %s",
+							 resp.getStatus(), resp));
+	    }
+	} catch (IOException e) {
+	    throw new RuntimeException("unexpected error occurred while communicating with prometheus server", e);
+	}
+    }
+
+    private float runSpEL(String query) {
```

Comment:
I know this is a private method but I think it will be good to either describe the basics of SpEL syntax or point to a resource it describes it. 

Currently when someone jumps into the development (or definition of queries) might not really know how to deal with SpEL.

Alternatively we could make it a part of our MED documentation (README file). This might be better because the SpEL covers both java code and an external query definition. If we find it a better approach then we could create a dedicated task to cover it and close this PR.

Reply:
Agreed that something like this should be added in the README.

Reply:
Created #29